test/tools/isoltest.cpp

```diff
@@ -60,17 +64,43 @@ struct TestStats
 	}
 };
 
+class TestFilter
+{
+public:
+	explicit TestFilter(string const& _filter): m_filter(_filter)
+	{
+		string filter{m_filter};
+
+		boost::replace_all(filter, "/", "\\/");
+		boost::replace_all(filter, "*", ".*");
+
+		m_filterExpression = regex{"(" + filter + "(\\.sol|\\.yul))"};
+	}
+
+	bool matches(string const& _name) const
+	{
+		return m_filter.empty() || regex_match(_name, m_filterExpression);
```

Comment:
Why is this done here and in `IsolTestOptions`?

Reply:
`IsolTestOptions` checks, if the value of `-t` is specified and valid filter expression (e.g. `semanticTests/*` or `semanticTests/array*` etc.). This checks if the filter is not set (all tests match) or if it's set and filename matches the filter.